packages/core/src/services/documentVersions/create.ts

```diff
@@ -36,32 +41,37 @@ function createDocument({
 }
 
 export async function createDocumentVersion({
+  documentUuid,
   projectId,
   name,
   commitUuid,
   documentType,
   parentId,
+  content,
 }: {
+  documentUuid?: string
   projectId: number
   name: string
   commitUuid: string
   documentType?: DocumentType
   parentId?: number
+  content?: string
 }) {
-  let commit = await findCommit({ uuid: commitUuid })
-  return Transaction.call<DocumentVersion>(async (tx) => {
-    if (!commit) {
-      const resultCommit = await createCommit({ projectId, db: tx })
-      if (resultCommit.error) return resultCommit
+  const commitResult = await findCommit({ commitUuid, projectId })
+  const commit = commitResult.unwrap()
 
-      commit = resultCommit.value
-    }
+  if (commit.mergedAt !== null) {
+    return Result.error(
+      new ForbiddenError('Cannot create a document version in a merged commit'),
+    )
+  }
```

Comment:
Added this restriction on `createDocumentVersion`.

Reply:
hmm would be cool to have model level validations like activerecord, @andresgutgon do u know if there's something for this in drizzle land?

Reply:
I saw this kind of repository pattern example:
https://gist.github.com/cayter/49d5c256a885d90c399ca6c1eca19f51

We could implement it and hook into `beforeCreate` to validate the row
https://gist.github.com/cayter/49d5c256a885d90c399ca6c1eca19f51#file-repository-ts-L357

But I didn't see anything official. 


I'm a fan of ActiveRecord but Drizzle is super functional. I don't think it is wrong to have a service to create documents with the validation inside. As long as we use it as unique way for creating documents